detectable/src/main/java/com/synopsys/integration/detectable/detectables/pip/poetry/PoetryDetectable.java

```diff
@@ -60,9 +71,28 @@ public DetectableResult applicable() {
         if (poetryLock == null && pyprojectToml == null) {
             return new FilesNotFoundDetectableResult(PYPROJECT_TOML_FILE_NAME, POETRY_LOCK);
         }
+        if (poetryLock == null && !containsToolDotPoetrySection(pyprojectToml)) {
+            return new SectionNotFoundDetectableResult(pyprojectToml.getName(), TOOL_DOT_POETRY_KEY);
+        }
         return new PassedDetectableResult();
     }
 
+    private boolean containsToolDotPoetrySection(@Nullable File pyprojectToml) {
```

Comment:
This really should move to a parser. I think TomlPoetrySectionParser with a PoetrySectionResult and it could  have a .wasFound(). 
So it could become something like `&& TomlPoetrySectionParser.parse(pyprojectToml).wasFound()`